### PR TITLE
recycler_in_fragment

### PR DESCRIPTION
**Pull request by PfcRico**
Data transfer with Fragment.newInstance(), code refactor for Adapter and CityDetails

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
 }
 
 android {
     compileSdk 31
 
+    buildFeatures {
+        viewBinding = true
+    }
+
+
```
Comment (atserkovnaya) [inline app/build.gradle:15]: please, remove extra line

File: app/src/main/java/com/example/weatherapp/data/network/ApiWeatherService.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            val okHttpClient = OkHttpClient.Builder()
+                .addInterceptor(requestInterceptor)
+                .addInterceptor(logger)
+                .build()
+
+            return Retrofit.Builder()
+                .client(okHttpClient)
+                .baseUrl("http://api.openweathermap.org/")
+                .addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(CoroutineCallAdapterFactory())
```
Comment (atserkovnaya) [inline app/src/main/java/com/example/weatherapp/data/network/ApiWeatherService.kt:52]: I guess this is outdate line

File: app/src/main/java/com/example/weatherapp/fragment/CityDetailsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        fun newInstance(cityName: String) = CityDetailsFragment().apply {
+            arguments = Bundle().apply {
+                putString("cityName", cityName)
+            }
+        }
+    }
+}
+
+
+
```
Comment (atserkovnaya) [inline app/src/main/java/com/example/weatherapp/fragment/CityDetailsFragment.kt:123]: extra lines)

File: app/src/main/java/com/example/weatherapp/fragment/CityDetailsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        return formatForDate.format(stringFormat)
+    }
+
+    private fun formatCurrentTime(date: Date): String {
+        return date.toString("yyyy/MM/dd HH:mm:ss")
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        arguments?.getString("cityName")?.let {
```
Comment (atserkovnaya) [inline app/src/main/java/com/example/weatherapp/fragment/CityDetailsFragment.kt:108]: Please, move "cityName" to constants and use in other places as constant

**Review by atserkovnaya — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (Jerika) [inline app/build.gradle:15]: Please, remove extra line

File: app/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Jerika) [inline app/src/main/res/values/styles.xml:1]: is this empty file really need?

File: app/src/main/res/layout/cities_list_fragment.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/recycler_view"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            />
+
+
```
Comment (Jerika) [inline app/src/main/res/layout/cities_list_fragment.xml:18]: empty line)

File: app/src/main/res/layout/city_view_fragment.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+<androidx.cardview.widget.CardView
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:card_view="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="@dimen/normal_x_6"
+    android:layout_marginTop="@dimen/normal_75"
+    android:layout_marginLeft="@dimen/normal_100"
+    android:layout_marginRight="@dimen/normal_100"
+    android:id="@+id/card_info"
+    card_view:cardBackgroundColor="#FFFF"
```
Comment (Jerika) [inline app/src/main/res/layout/city_view_fragment.xml:11]: please, move to colors.xml

**Review by Jerika — CHANGES_REQUESTED** [review]

**Review by Jerika — APPROVED** [review]

**Pull request merged**